### ISSUE TITLE
Vision only weights

### ISSUE THREAD

**Issue by Lewis-Lu** [haomo-ai/SuperFusion#5]
Hi there,

is there any standalone .pth weights for vision-only model ?
Or is it posssible that I filter out the LiDAR-related weight from original .pth for an acceptable results ?

Best,
Lewis

**Comment by donghao51** [comment]
hello, you can change lidar=False in this line to disable using LiDAR (https://github.com/haomo-ai/SuperFusion/blob/a2671f05e7e18c9ab981bcb18812c400d284c545/model_front/__init__.py#L6). 

however, the bev features are trained using the concatenation of lidar and vision features. You can either set all lidar bev features to 0 or finetune layers after bev alignment module.

**Comment by Lewis-Lu** [comment]
Hi Donghao,

Thanks, I have tried this before and obviously caught the following error.

`size mismatch for bevencode.conv1.weight: copying a param with shape torch.Size([64, 192, 7, 7]) from checkpoint, the shape in current model is torch.Size([64, 64, 7, 7]).`

Thus, from your comment, I might create a LiDAR feature filled with zero to try, if I understand it right. I would try it. If so, how do we deal with the image backbone input as it's concated a projected depth info. My thought was to use depth filled with zero for inference according to your comment.

I am not sure that zero-value would affect the perf or not.

Futhermore, did you consider only using LiDAR for supervision, then only using Camera, with only image input, during Inference ? If so, that would be great to have such .pth. 

Best,
Lewis

**Comment by Lewis-Lu** [comment]
Hi Donghao,

I have tried the method as mentioned. The 0-valued LiDAR feature is non-doable. It definitely needs fine-tuning.

softmax(segmentation) cannot extract the much feature compared to that of the fusioned one.
![image](https://github.com/haomo-ai/SuperFusion/assets/24478504/895ed201-96fa-4783-828e-6ce39577a371)

Best,
Lewis
